packages/back-end/src/controllers/stripe.ts

```diff
@@ -4,141 +4,155 @@ import {
-    },
-  });
-}
-export async function postStartTrial(
-  req: AuthRequest<{ qty: number; name: string }>,
+type PriceData = {
+  [key: string]: Stripe.Price;
+};
+
+const discountData: DiscountData = {};
+
+const priceData: PriceData = {};
+
+export async function postNewSubscription(
+  req: AuthRequest<{ qty: number }>,
   res: Response
 ) {
-  const { qty, name } = req.body;
+  const { qty } = req.body;
 
-  // If user already has a subscription, return immediately
-  if (req.organization?.subscription?.id) {
-    return res.status(200).json({
-      status: 200,
-    });
+  req.checkPermissions("organizationSettings");
+
+  const { org } = getOrgFromReq(req);
+
+  if (!org) {
+    throw new Error("No organization found");
   }
 
-  try {
-    // Create organization first if needed
-    if (!req.organization) {
-      if (name.length < 3) {
-        throw new Error("Company name must be at least 3 characters long");
-      }
-      req.organization = await createOrganization(
-        req.email || "",
-        req.userId || "",
-        name,
-        ""
-      );
-    }
+  const desiredQty = getNumberOfMembersAndInvites(org);
 
-    // Create customer in Stripe if not exists
-    if (!req.organization.stripeCustomerId) {
-      const resp = await stripe.customers.create({
-        email: req.email || "",
-        name: req.name || "",
-        metadata: {
-          user: req.userId || "",
-          organization: req.organization.id,
-        },
-      });
-      req.organization.stripeCustomerId = resp.id;
-
-      await updateOrganization(req.organization.id, {
-        stripeCustomerId: resp.id,
-      });
-    }
+  if (desiredQty !== qty) {
+    throw new Error(
+      "Number of users is out of date. Please refresh the page and try again."
+    );
+  }
 
-    // Start subscription trial without payment method
-    const subscription = await stripe.subscriptions.create({
-      customer: req.organization.stripeCustomerId,
-      coupon: STRIPE_DEFAULT_COUPON,
-      collection_method: "charge_automatically",
-      trial_from_plan: true,
+  let stripeCustomerId: string;
+
+  if (org.stripeCustomerId) {
+    stripeCustomerId = org.stripeCustomerId;
+  } else {
+    const { id } = await stripe.customers.create({
       metadata: {
-        user: req.userId || "",
-        organization: req.organization.id,
+        growthBookId: org.id,
+        ownerEmail: org.ownerEmail,
       },
-      items: [
-        {
-          price: STRIPE_PRICE,
-          quantity: qty,
-        },
-      ],
+      name: org.name,
     });
+    stripeCustomerId = id;
 
-    // Save in Mongo
-    await updateSubscription(subscription);
+    await updateOrganization(org.id, {
+      stripeCustomerId: stripeCustomerId,
+    });
+  }
 
-    res.status(200).json({ status: 200 });
-  } catch (e) {
-    res.status(400).json({
-      status: 400,
-      message: e.message,
+  const session = await stripe.checkout.sessions.create({
```

Comment:
Do we want a check to make sure they don't already have an active subscription before starting a new checkout session?

Reply:
I'm happy to add it as a safe guard. No one SHOULD be able to get into this with an active subscription, but crazier things have definitely happened.

Reply:
The code relies on the front-end only calling this API endpoint in specific circumstances.  Front-end code in a browser is notoriously unreliable (extensions, people messing around with dev tools, etc.), so it's best to have safeguards in the back-end just in case.